content/articles/creating-a-shopping-cart-api/index.md

```diff
@@ -0,0 +1,592 @@
+3. Spring boot-> don't use the one with a snapshot, better still use the default 
+4. Group-> shopping
+5. Artifact->cart
+6. Name-> cart
+7. Description-> it is for you to explain briefly what your project is all about but for now we will leave it like that.
+8. Dependencies-> Search for the following in the search bar and add them to your dependencies for this project.
+- Lombok-> for your annotations like @Data for your entities and prevent you from writing your setters and getters and allows you to Autowire.
+- Spring web -> necessary when building a web app
+- devtool-> For automatic reloading
+9. Packaging -> Jar
+10.Java-> 11 or anyone you choose, better still choose the default.
+### Creating Your Entities/Models
+Here you create the entities that will relate to the database or the data structure you are mapping your object to if you are not using a database. Before we view the code base of creating our entities let's view what the structure of the entire application will look like![Package Structure](shoppingapplication.JPG)
+Here you can see the various classes in the package
+![Class Structure](shoppingcart2.JPG).
+In our models we have the following classes:
+- Admin
+- Customer
+- Product
+### Admin
+```java
+@Data
+public class Admin {
+    private Long id;
+    private String firstName;
+    private String lastName;
+
+    public Admin(Long id, String firstName, String lastName) {
+        this.id = id;
+        this.firstName = firstName;
+        this.lastName = lastName;
+    }
+}
+
+}
+```
+### Customer
+```java 
+@Data
+public class Customer {
+
+    private Long id;
+   private String firstName;
+   private String lastName;
+   private BigDecimal Balance;
+
+
+    public Customer(Long id, String firstName, String lastName, BigDecimal balance) {
+        this.id = id;
+        this.firstName = firstName;
+        this.lastName = lastName;
+        Balance = balance;
+    }
+}
+```
+### Product
+```java
+@Data
+public class Product {
+    private String Id;
+    private String productName;
+    private BigDecimal cost;
+
+    public Product(String id, String productName, BigDecimal cost) {
+        Id = id;
+        this.productName = productName;
+
+        this.cost = cost;
+    }
+
+    public Product() {
+
+    }
+
+    @Override
+    public String toString() {
+        return "Product{" +
+                "Id='" + Id + '\'' +
+                ", productName='" + productName + '\'' +
+                ", cost=" + cost +
+                '}';
+    }
+}
+```
+In the model package, you should have these classes in it. These classes or entities provide the feature or attribute to what it represents.
+### Creating Your Repository
+The importance of creating a repository is to map your entity to the database, It's like a store and also maps your entities. In case you want to work on your service instead of going looking for how to get some element your repository helps you get it. Always make sure you use Autowire. 
```

Comment:
> In case you want to work on your service instead of going looking for how to get some element your repository helps you get it.

The wording here sounds off. While I think I know what you are trying to say here, the way you are saying it may not be clear. Please revise it.